src/python/pants/backend/shell/target_types.py

```diff
@@ -442,6 +461,41 @@ class ShellCommandTarget(Target):
     )
 
 
+class ShellRunInSandboxTarget(Target):
+    alias = "experimental_run_in_sandbox"
+    core_fields = (
+        *COMMON_TARGET_FIELDS,
+        RunInSandboxRunnableField,
+        RunInSandboxArgumentsField,
+        ShellCommandExecutionDependenciesField,
+        ShellCommandOutputDependenciesField,
+        ShellCommandLogOutputField,
+        ShellCommandOutputsField,
+        RunInSandboxSourcesField,
+        ShellCommandTimeoutField,
+        ShellCommandToolsField,
+        ShellCommandExtraEnvVarsField,
+        EnvironmentField,
+    )
+    help = softwrap(
+        """
+        Execute any runnable target for its side effects.
+
+        Example BUILD file:
+
+            experimental_run_in_sandbox(
+                runnable=":python_source",
+                argv=[""],
+                tools=["tar", "curl", "cat", "bash", "env"],
```

Comment:
Not directly related to this change, but I wonder if this tools metadata should/could be some sort of dependency of (runnable) targets. Thus, users of that file would be able to automatically pull in the tools, rather than potentially needing to repeat them in all `esc`/`experimental_run_in_sandbox` etc. invocations that use them? 

(Presumably `esc` with its inline command would still need to support explicit specification too)

For example:

```python
shell_sources(
    name="scripts",
    system_tools=["bash", ...],
)

python_source(
    source="whatever.py", # includes subprocess.run(["some-tool", ...])
    system_tools=["some-tool"],
    dependencies=[":scripts"]
)

# PATH for execution will automatically include bash, some-tool, etc.
experimental_run_in_sandbox(
    runnable="./whatever.py"
    outputs=[...]
)
```

Reply:
That's fascinating, and would be a fairly trivial change. Would you mind popping this out into a separate ticket and assigning me to it?